_opensearch/rest-api/snapshots/create-repository.md

```diff
@@ -0,0 +1,96 @@
+---
+layout: default
+title: Register Snapshot Repository
+parent: Snapshot APIs
+grand_parent: REST API reference
+nav_order: 1
+---
+
+## Register or update snapshot repository
+
+Registers a new repository to store snapshots, or updates information for an existing repository. Repositories can be of two types:
+
+* File system (`fs`)
+
+* Amazon S3 bucket (`s3`)
+
+To learn more about repositories, see [Register repository]({{site.url}}{{site.baseurl}}/opensearch/snapshots/snapshot-restore#register-repository).
+
+
+### Path parameters
+
+Parameter | Data Type | Description
+:--- | :--- | :---
+repository | String | Repostory name. |
+
+#### Sample request
+
+You can register repositories with `POST` or `PUT`. 
+{: .note}
+
+The following requests register a repository called `my-first-repo`:
+
+`POST _snapshot/my-first-repo/` registers a snapshot repository. Requires a request body as described in [Request body](#request-body).
+
+`PUT _snapshot/my-first-repo/` registers a snapshot repository if it does not exist; otherwise, updates information about the repository. For creation, requires a request body as described in [Request body](#request-body).
+
+To verify that the repository was registered, use the [Get snapshot repository API]({{site.url}}{{site.baseurl}}/opensearch/rest-api/snapshots/get-snapshot-repository), passing the repository name as the `repository` path parameter.
+{: .note}
+
+### Request body
```

Comment:
Please call this section "Request fields" and move it to before the "Sample request" section

Reply:
I have made this change.